FGEGraphics/UISystem/UIInputLabel.cs

```diff
@@ -0,0 +1,469 @@
+﻿//
+// This file is part of the Frenetic Game Engine, created by Frenetic LLC.
+// This code is Copyright (C) Frenetic LLC under the terms of a strict license.
+// See README.md or LICENSE.txt in the FreneticGameEngine source root for the contents of the license.
+// If neither of these are available, assume that neither you nor anyone other than the copyright holder
+// hold any right or permission to use this software until such time as the official license is identified.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FGECore.CoreSystems;
+using FGECore.MathHelpers;
+using FGEGraphics.ClientSystem;
+using FGEGraphics.GraphicsHelpers;
+using FGEGraphics.GraphicsHelpers.FontSets;
+using FGEGraphics.UISystem.InputSystems;
+using OpenTK.Mathematics;
+using OpenTK.Windowing.GraphicsLibraryFramework;
+
+namespace FGEGraphics.UISystem;
+
+/// <summary>Represents an editable text area.</summary>
+// TODO: Text alignment
+// TODO: Horizontal scrolling within bounds, with option
+// TODO: Vertical scrolling when bounds exceeded
+// TODO: Cap text length
+public class UIInputLabel : UIClickableElement
```

Comment:
`UIInputLabel` is a confusing name - is this a Text input box?

Reply:
Yeah -- there's a `UIInputBox` already but I don't want to replace it just yet